akka-http-tests/src/test/java/akka/http/javadsl/server/JavaRouteTest.java

```diff
@@ -0,0 +1,351 @@
+package akka.http.javadsl.server;
+
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.junit.Test;
+
+import akka.http.javadsl.model.ContentType;
+import akka.http.javadsl.model.ContentTypes;
+import akka.http.javadsl.model.HttpCharsets;
+import akka.http.javadsl.model.HttpEntity;
+import akka.http.javadsl.model.HttpRequest;
+import akka.http.javadsl.model.HttpResponse;
+import akka.http.javadsl.model.MediaRanges;
+import akka.http.javadsl.model.MediaTypes;
+import akka.http.javadsl.model.RequestEntity;
+import akka.http.javadsl.model.StatusCodes;
+import akka.http.javadsl.model.headers.Accept;
+import akka.http.javadsl.model.headers.RawHeader;
+import akka.http.javadsl.testkit.JUnitRouteTest;
+//FIXME discuss how to provide a javadsl.CustomHeader where render() is either pre-implemented or trivial to write in Java
+import akka.http.scaladsl.model.headers.CustomHeader;
+import akka.japi.pf.PFBuilder;
+import akka.util.ByteString;
+import static akka.http.javadsl.server.StringUnmarshaller.*;
+import static akka.http.javadsl.server.PathMatchers.*;
```

Comment:
I'm not against static methods. I just wouldn't import them statically in tests/docs. ;)